Functions/src/main/java/com/epam/engx/cleancode/functions/task2/SitePage.java

```diff
@@ -20,10 +16,23 @@ public SitePage(String siteGroup, String userGroup) {
     }
 
     public String getEditablePageUrl(Map<String, String> params) {
-        String paramsString = "";
+        return composeUrl(params);
+    }
+
+    private String composeUrl(Map<String, String> params) {
+        StringBuilder stringBuilder = new StringBuilder(URL_OF_EDITABLE);
+        addParameters(params, stringBuilder);
+        addAttributes(stringBuilder);
+        return stringBuilder.toString();
+    }
+
+    private void addParameters(Map<String, String> params, StringBuilder stringBuilder) {
         for (Map.Entry<String, String> param : params.entrySet())
-            paramsString += "&" + param.getKey() + "=" + param.getValue();
-        return HTTP + DOMAIN + EDITABLE + paramsString + getAttributes();
+            stringBuilder.append("&").append(param.getKey()).append("=").append(param.getValue());
+    }
+
+    private void addAttributes(StringBuilder stringBuilder) {
+        stringBuilder.append(getAttributes());
```

Comment:
Look at line 39. Hidden string concatenation here. So change `getAttributes` to return SB instead of String

Reply:
https://www.deepcode.ai/app/gh/Yauhenda/refactoring_tasks/169c5b013f301ecf2bae5c7bde6a45d5ef61f6aa/Yauhenda/refactoring_tasks/9ddfd0e930d2b30daa4fd3bda0196b5e8c20ecd8/_/%2FFunctions%2Fsrc%2Fmain%2Fjava%2Fcom%2Fepam%2Fengx%2Fcleancode%2Ffunctions%2Ftask2%2FSitePage.java/java%2Fdc%2FNoStringConcatArg1~append/35/code/